router/transformer/transformer.go

```diff
@@ -150,11 +156,52 @@ func (trans *HandleT) Transform(transformType string, transformMessage *types.Tr
 	return destinationJobs
 }
 
+func (trans *HandleT) ResponseTransform(responseData integrations.DeliveryResponseT, destName string) (statusCode int, respBody string) {
+	rawJSON, err := json.Marshal(responseData)
+	if err != nil {
+		panic(err)
+	}
+	var resp *http.Response
+	var respData []byte
+	url := getResponseTransformURL(destName)
+	trans.transformerResponseTransformRequestTime.Start()
+	resp, err = trans.client.Post(url, "application/json; charset=utf-8", bytes.NewBuffer(rawJSON))
+	trans.transformerResponseTransformRequestTime.End()
+	if resp != nil && resp.Body != nil {
+		respData, _ = io.ReadAll(resp.Body)
+	}
+	integrations.CollectIntgErrorStats([]byte(gjson.GetBytes(respData, "output").Raw), false)
+	var contentTypeHeader string
+	if resp != nil && resp.Header != nil {
+		contentTypeHeader = resp.Header.Get("Content-Type")
+	}
+	if contentTypeHeader == "" {
+		//Detecting content type of the respBody
+		contentTypeHeader = http.DetectContentType(respData)
+	}
+	//If content type is not of type "*text*", overriding it with empty string
+	if !(strings.Contains(strings.ToLower(contentTypeHeader), "text") ||
+		strings.Contains(strings.ToLower(contentTypeHeader), "application/json") ||
+		strings.Contains(strings.ToLower(contentTypeHeader), "application/xml")) {
+		respData = []byte("")
+	}
+	if err != nil {
+		respData = []byte("")
+		trans.logger.Errorf("[Transformer Response Transform request] :: %+v", err)
+		return http.StatusInternalServerError, string(respData)
+	}
+	resp.Body.Close()
+	return resp.StatusCode, string(respData)
+}
+
+//is it ok to use same client for network and transformer calls? need to understand timeout setup in router
 func (trans *HandleT) Setup() {
 	trans.logger = pkgLogger
 	trans.tr = &http.Transport{}
-	trans.client = &http.Client{Transport: trans.tr}
+	trans.client = &http.Client{Transport: trans.tr, Timeout: 10 * time.Minute}
```

Comment:
I don't think we would need a specific timeout here. Let's remove it

Reply:
This is the timeout for requests to transformers. When we were doing proxying we kept it 10min for safety as we there could be long requests to destinations. Now we can remove this/ reduce it to much smaller durationion

Reply:
We should ideally restrict any http timeouts to not more than 30s. Probably can have it configured for destinations which by default take longer

Can we add it as a configurable variable if it makes sense

Reply:
As discussed with @sanpj2292  we will for now use default timeout of go-HTTP-client and and not configure explicitly if required we can do it as part of enhancement

Reply:
Go HTTP client does not have an explicit timeout I guess and requests can get stuck for a while